src/Client.php

```diff
@@ -0,0 +1,235 @@
+     *
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * Constructs a new Client instance.
+     *
+     * @param string      $tulipUrl
+     * @param string      $apiVersion
+     * @param string|null $clientId
+     * @param string|null $sharedSecret
+     */
+    public function __construct($tulipUrl, $apiVersion = '1.1', $clientId = null, $sharedSecret = null)
+    {
+        $this->tulipUrl = $tulipUrl;
+        $this->apiVersion = $apiVersion;
+        $this->clientId = $clientId;
+        $this->sharedSecret = $sharedSecret;
+    }
+
+    /**
+     * Returns the full Tulip API URL for the specified service and action.
+     *
+     * @param string $serviceName
+     * @param string $action
+     *
+     * @return string
+     */
+    public function getServiceUrl($serviceName, $action)
+    {
+        return sprintf('%s/api/%s/%s/%s', $this->tulipUrl, $this->apiVersion, $serviceName, $action);
+    }
+
+    /**
+     * Set the HTTP client instance.
+     *
+     * @param ClientInterface $httpClient
+     */
+    public function setHTTPClient(ClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * Call a Tulip API service.
+     *
+     * @param string $serviceName
+     * @param string $action
+     * @param array  $parameters
+     * @param array  $files
+     *
+     * @return ResponseInterface
+     *
+     * @throws RequestException
+     */
+    public function callService($serviceName, $action, array $parameters = array(), array $files = array())
+    {
+        $httpClient = $this->getHTTPClient();
+
+        $url = $this->getServiceUrl($serviceName, $action);
+        $request = new Request('POST', $url, $this->getRequestHeaders($url, $parameters), $this->getRequestBody($parameters, $files));
+
+        try {
+            $response = $httpClient->send($request);
+        } catch (GuzzleRequestException $exception) {
+            $response = $exception->getResponse();
+        }
+
+        $this->validateAPIResponseCode($request, $response);
+
+        return $response;
+    }
+
+    /**
+     * Returns a ClientInterface instance.
+     *
+     * @return ClientInterface
+     */
+    private function getHTTPClient()
+    {
+        if ($this->httpClient instanceof ClientInterface === false) {
+            $this->httpClient = new HTTPClient();
+        }
+
+        return $this->httpClient;
+    }
+
+    /**
+     * Returns the request authentication headers when a client ID and shared secret are provided.
+     *
+     * @param string $url
+     * @param array  $parameters
+     *
+     * @return array
+     */
+    private function getRequestHeaders($url, array $parameters)
+    {
+        $headers = array();
+
+        if (isset($this->clientId) && isset($this->sharedSecret)) {
+            $objectIdentifier = null;
+            if ($this->apiVersion === '1.1' && isset($parameters['id'])) {
+                $objectIdentifier = $parameters['id'];
```

Comment:
This library is merely implementing the API. :wink:

Besides that version 1.1 is not meant as a replacement, just an alternative. That's why it wasn't versioned as 2.0. Both API versions are still available.